inference-engine/src/gna_plugin/transformations/convert_padded2valid_conv.cpp

```diff
@@ -0,0 +1,471 @@
+    std::shared_ptr<opset1::MaxPool> max_pool;
+    std::shared_ptr<op::util::UnaryElementwiseArithmetic> af;
+    std::shared_ptr<Node> bias_const;
+    std::shared_ptr<Node>last_op_in_sequence_for_replacement;
+    bool disable_nhwc_to_nchw_option;
+};
+
+struct ConvData {
+    size_t input_height;
+    size_t input_width;
+    size_t input_channel_count;
+    size_t filter_height;
+    size_t filter_width;
+    size_t filter_count;
+    size_t filter_dilation_x;
+    size_t filter_dilation_y;
+    size_t filter_stride_x;
+    size_t filter_stride_y;
+    size_t pads_begin_y;
+    size_t pads_begin_x;
+    size_t pads_end_y;
+    size_t pads_end_x;
+    op::PadType padding_type;
+    size_t output_channel_count;
+    Shape output_shape;
+};
+
+bool TransposeOrderMatches(std::shared_ptr<Transpose> transpose, std::vector<int64_t> order) {
+    if (!transpose)
+        return false;
+    const Output<Node>& transpose_order = transpose->input_value(1);
+    auto transpose_order_dim = transpose_order.get_shape().size();
+
+    if (transpose_order_dim != 1 || transpose_order.get_shape()[0] != order.size())
+        return false;
+
+    auto const_with_order_values = std::dynamic_pointer_cast<opset1::Constant>(transpose_order.get_node_shared_ptr());
+    if (!const_with_order_values)
+        return false;
+
+    const int64_t* data = const_with_order_values->get_data_ptr<int64_t>();
+    if (!data)
+        return false;
+
+    for (size_t i = 0; i < order.size(); i++) {
+        if (order[i] != data[i])
+            return false;
+    }
+
+    return true;
+}
+
+std::shared_ptr<opset1::StridedSlice> FlatCrop(Output<Node> input, size_t offset, size_t size) {
+    return std::make_shared<opset1::StridedSlice>(
+        input, // data
+        opset1::Constant::create(element::i64, Shape{ 2 }, { (size_t)0, offset }), // begin sice index
+        opset1::Constant::create(element::i64, Shape{ 2 }, { (size_t)0, offset + size }), // end slice index
+        opset1::Constant::create(element::i64, Shape{ 2 }, { (size_t)1, (size_t)1 }), // strides
+        std::vector<int64_t>{1, 0},  // begin mask
+        std::vector<int64_t>{1, 0}); // end mask
+}
+
+template<class T>
+bool VerifyLayer(std::shared_ptr<T> layer) {
+    auto layer_output_0 = layer->get_output_target_inputs(0);
+    return layer_output_0.size() == 1 ? true : false;
+}
+
+template<>
+bool VerifyLayer<>(std::shared_ptr<opset1::MaxPool> max_pool) {
+    auto layer_output_0 = max_pool->get_output_target_inputs(0);
+    if (layer_output_0.size() != 1)
+        return false;
+
+    auto pool_strides = max_pool->get_strides();
+    auto pool_kernel = max_pool->get_kernel();
+
+    // Check if MaxPool vertical stride == pool size (TODO: remove when 50386 and 50379 are fixed)
+    // Check if padding is VALID
+    if (max_pool->get_auto_pad() != PadType::VALID ||
+        pool_kernel.size() != 2 || pool_strides.size() != 2 ||
+        pool_kernel[0] != pool_strides[0] || pool_kernel[0] > 8)
```

Comment:
8 ?

Reply:
This can be verified when the scenarios with maxpool are unblocked. I added a description in a TODO line.